tools/auto_fill_stable_diffusion_cache.py

```diff
@@ -0,0 +1,192 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to cache models for stable diffusion."""
+import argparse
+import json
+import logging
+import re
+import subprocess
+import tempfile
+import time
+
+import requests
+from huggingface_hub import login
+
+from optimum.neuron import version as optimum_neuron_version
+
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, force=True)
+logger = logging.getLogger()
+
+
+def get_neuronx_cc_version():
+    result = subprocess.run(["neuronx-cc", "--version"], capture_output=True, text=True)
+    version_match = re.search(r"NeuronX Compiler version ([\d\.]+\+[a-f0-9]+)", result.stderr)
+    if version_match:
+        return version_match.group(1)
+    else:
+        raise ValueError("Version information not found in the output")
```

Comment:
I think we have already a function that does that in `optimum/neuron/utils/version_utils.py`.

Reply:
Yeah, it is, for the script, I just replicated https://github.com/huggingface/optimum-neuron/blob/main/tools/auto_fill_inference_cache.py. I will replace with our already implemented function there and fuse these two scripts.